src/libraries/System.Security.Cryptography.Algorithms/tests/DefaultECDiffieHellmanProvider.Android.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Security.Cryptography.EcDiffieHellman.Tests
+{
+    public partial class ECDiffieHellmanProvider : IECDiffieHellmanProvider
+    {
+        public bool IsCurveValid(Oid oid)
+        {
+            if (PlatformDetection.IsOSXLike)
```

Comment:
I guess we did runtime checks for OSX in the Unix implementation instead of having separate files/builds - would it make sense to do the same for Android?

Reply:
Either way's OK, but since this makes an interop call it's probably cleaner to have build-time separation.